pandas/core/tools/datetimes.py

```diff
@@ -687,7 +688,7 @@ def to_datetime(
     yearfirst: bool = False,
     utc: bool = False,
     format: str | None = None,
```

Comment:
Would making `format` default to `mixed` be "less" API breaking behavior compared to what `to_datetime` did in 1.5?

Reply:
Yes, I did consider it, but then realised it would bring back the decade-long bug which PDEP4 was meant to address 😄 
```python
In [10]: to_datetime(['12/01/2000 00:00', '13/01/2000 00:00'])
Out[10]: DatetimeIndex(['2000-12-01', '2000-01-13'], dtype='datetime64[ns]', freq=None)
```

The US is the only country in the world where month-first is the default, `format='mixed'` by default results in unexpected behaviour for people in the vast majority of the world

I'm OK with allowing flexible parsing, but really think it should be opt-in